### PR TITLE
B 17147 loa parser

### PR DESCRIPTION
**Pull request by cameroncaci**
## Summary

Added Line Of Accounting parsing package to receive a .txt file just like the Transportation Account package previously created and PR'd. This package however has been revised to use the new struct provided by Truss.

## Verification Steps for Reviewers

- Proper pointer handling
-  Coding standards
-  Descriptive comments

### How to test

N/A - Awaiting upstream dependencies

### CODE REVIEW

File: pkg/models/transportation_accounting_code_test.go [hunk 1]
```diff
@@ -26,7 +99,7 @@ func (suite *ModelSuite) Test_InvalidTac() {
 	suite.NoError(err)
 }
 
-func (suite *ModelSuite) Test_CanSaveAndFetchTac() {
+func (suite *ModelSuite) TestCanSaveAndFetchTac() {
```
Comment (j-Rosenthal) [inline pkg/models/transportation_accounting_code_test.go]: I'm wondering if renaming the methods will cause conflicts when merging back with Truss' main branch. 

I know that the correct naming convention is `Test****` not `Test_****` so I am okay with the renaming here. (just food for thought.)

File: pkg/parser/loa/parse.go [hunk 2]
```diff
@@ -0,0 +1,221 @@
...
+	"io"
+	"reflect"
+	"strconv"
+	"strings"
+	"time"
+
+	"github.com/transcom/mymove/pkg/models"
+)
+
+const timeConversionMethod = "2006-01-02 15:04:05"
```
Comment (j-Rosenthal) [inline pkg/parser/loa/parse.go:16]: Where does this specific time and date come from?

File: pkg/models/transportation_accounting_code_test.go (hunk 1 above)
Comment (cameroncaci) [reply pkg/models/transportation_accounting_code_test.go]: That's a good point - I don't see any conflicts when running `make server_test` though. @deandreJones thoughts?

File: pkg/parser/loa/parse.go (hunk 2 above)
Comment (cameroncaci) [reply pkg/parser/loa/parse.go:16]: Mimicking from here: https://go.dev/src/time/format.go - it serves as the format for parsing

File: pkg/models/transportation_accounting_code_test.go (hunk 1 above)
Comment (deandreJones) [reply pkg/models/transportation_accounting_code_test.go]: looking through other test classes, i come across some functions as Test**** and others Test_****
so i dont see any reason to make any changes here, esp if tests are running and passing

Comment (cameroncaci) [reply pkg/models/transportation_accounting_code_test.go]: Just pushed a new commit reverting their test names back to "Test_", but I maintained my "Test***" syntax within the same file
@deandreJones @j-Rosenthal

**Review by j-Rosenthal — DISMISSED** [review]

**Review by deandreJones — DISMISSED** [review]

**Review by TevinAdams — DISMISSED** [review]

**Comment by deandreJones** [comment]
conflicts

**Review by danieljordan-caci — DISMISSED** [review]

**Review by j-Rosenthal — APPROVED** [review]

**Review by deandreJones — APPROVED** [review]

**Pull request merged**